### PR TITLE
Support different ccache versions to show ccache statistics

### PR DESCRIPTION
**Pull request by eukarpov**

### CODE REVIEW

File: .github/scripts/config.sh [hunk 1]
```diff
@@ -80,4 +80,11 @@ PATH="$PATH:$TOOLCHAIN_PATH/bin"
 if [[ "$CCACHE" = 1 ]]; then
     PATH=$CCACHE_LIB_DIR:$TOOLCHAIN_CCACHE_LIB_DIR:$PATH
     export CCACHE_DIR=$CCACHE_DIR_PATH
+    set +o pipefail
+    if ccache -sv 2>&1 | grep -q "invalid option"; then
```
Comment (Blackhex) [inline .github/scripts/config.sh:84]: I'd move this to https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/blob/main/.github/scripts/enable-ccache.sh

File: .github/scripts/build.sh [hunk 2]
```diff
@@ -28,7 +28,7 @@ if [[ "$RUN_BOOTSTRAP" = 1 || "$RESET_SOURCES" = 1 ]]; then
 fi
 
 if [[ "$CCACHE" = 1 ]]; then
-    ccache -sv
+    $CCACHE_STATISTICS
```
Comment (Blackhex) [inline .github/scripts/build.sh:31]: I'd prefer `ccache $CCACHE_STATISTICS`.

File: .github/scripts/config.sh [hunk 3]
```diff
@@ -80,4 +80,11 @@ PATH="$PATH:$TOOLCHAIN_PATH/bin"
 if [[ "$CCACHE" = 1 ]]; then
     PATH=$CCACHE_LIB_DIR:$TOOLCHAIN_CCACHE_LIB_DIR:$PATH
     export CCACHE_DIR=$CCACHE_DIR_PATH
+    set +o pipefail
```
Comment (Blackhex) [inline .github/scripts/config.sh:83]: Invoking a sub-shell is safer and there is no need to restore the setting:
```bash
$ (set -o pipefail && command1 | command2 | command3)
```

**Review by Blackhex — CHANGES_REQUESTED** [review]

File: .github/scripts/config.sh (hunk 3 above)
Comment (eukarpov) [reply .github/scripts/config.sh:83]: Yes, it was my initial idea. Somehow I decided to do in different way. I will change it.

File: .github/scripts/config.sh (hunk 1 above)
Comment (eukarpov) [reply .github/scripts/config.sh]: Changed.

File: .github/scripts/build.sh (hunk 2 above)
Comment (eukarpov) [reply .github/scripts/build.sh]: Changed.

File: .github/scripts/config.sh (hunk 3 above)
Comment (eukarpov) [reply .github/scripts/config.sh]: Not needed anymore after refactoring.

File: .github/scripts/config.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (eukarpov) [reply .github/scripts/config.sh]: Added.

**Review by Blackhex — APPROVED** [review]
Looks much better now.

**Pull request merged**